src/components/ProjectSettingsTabs/PluginsTab/transformers.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  ProjectInput,
+  ProjectSettingsInput,
+  ProjectSettingsQuery,
+  RepoSettingsQuery,
+} from "gql/generated/types";
+import { FormState } from "./types";
+
+export const gqlToForm = (
+  data:
+    | ProjectSettingsQuery["projectSettings"]
+    | RepoSettingsQuery["repoSettings"]
+): FormState => {
+  if (!data) return null;
+
+  const { projectRef } = data;
+  return {
+    performanceSettings: {
+      perfEnabled: projectRef.perfEnabled,
+    },
+    buildBaronSettings: {
+      ticketCreateProject: projectRef.buildBaronSettings.ticketCreateProject,
+      ticketSearchProjects: projectRef.buildBaronSettings.ticketSearchProjects
+        ? projectRef.buildBaronSettings.ticketSearchProjects.map(
+            (searchProject) => ({ searchProject })
+          )
+        : [],
```

Comment:
[nit] Just a lil syntactic sugar
```suggestion
      ticketSearchProjects:
        projectRef?.buildBaronSettings?.ticketSearchProjects?.map(
            (searchProject) => ({ searchProject })
          ) ?? [],
```